msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
This is over-complicated I think. It's sufficient to specify that the ``branch_colours`` is a dictionary, indexed by the younger node. Any use cases for the array input can be trivially remapped to a dict. The consensus colour thing seems far too complicated to me.